src/WebJobs.Script.WebHost/Extensions/ExceptionExtensions.cs

```diff
@@ -21,6 +21,12 @@ public static (string exceptionType, string exceptionMessage, string exceptionDe
                 innerException = innerException.InnerException;
             }
 
+            // If innerMost exception does not have the stack trace, default to the outermost one.
+            if (innerException.StackTrace == null || innerException.StackTrace.Length < 1)
+            {
+                innerException = exception;
+            }
+
             string exceptionType = innerException.GetType().ToString();
             string exceptionMessage = Sanitizer.Sanitize(innerException.Message);
             string exceptionDetails = Sanitizer.Sanitize(innerException.ToFormattedString());
```

Comment:
What if, instead, we passed the outer-most exception into `ToFormattedString()`? It looks like it walks through the inner exceptions to write out everything...

Reply:
Yeah, I thought about that but I wasn't sure if that would introduce any logging regressions around exception. But ideally, the outermost exception has the most amount of stack trace so that would have more info. Okay to risk it ?